magic-modules/branch-magic-modules.sh

```diff
@@ -0,0 +1,12 @@
+#! /bin/bash
+set -e
+set -x
+
+pushd "magic-modules"
+BRANCH=$(git rev-parse --short HEAD)
+git checkout -B "$BRANCH"
```

Comment:
Why do we need to create a branch? As line 10 indicates this folder cannot be shared with other runs (as you write the branch name there). So why not simply using HEAD instead or forget branches altogether?

Reply:
It's not shared between other runs at all - virtually nothing in Concourse can really be shared between other runs, except identical `get`s.

We need the branch because the altered magic-modules repo is going to be pushed to github, and we want to push it somewhere where it won't get in the way of development.  The branch named after the commit hash is both guaranteed-reasonably-unique and recomputable.  We need to store the branch because we use the same branch for the intermediate submodules.

It needs to be pushed to github because concourse doesn't have any storage, so we need some place to put it so we can fan out to run the tests on it.

Reply:
Ah, got it. Thanks.